src/Workspaces/Core/Portable/Rename/RenameLocation.cs

```diff
@@ -69,5 +64,118 @@ public override int GetHashCode()
+                        return true;
+
+                    case CandidateReason.NotReferencable:
+                        // Happens when the user does something like directly accessing
+                        // the accessor of a normal property.  In this case, we do still
+                        // want to allow the rename to happen.
+                        return true;
+
+                    case CandidateReason.Inaccessible:
+                        // Can trivially occur in code that is in an initially broken state
+                        // where inaccessible members are being referenced.  We still want
+                        // to update these references.
+                        return true;
+
+                    case CandidateReason.NotAValue:
+                    case CandidateReason.NotAVariable:
+                        // Happens with code like "NS = 1".  If "NS" is binding now to a 
+                        // namespace, then it's likely something has gone very wrong (similar to
+                        // NotATypeOrNamespace), and we shouldn't update this reference.
+                        return false;
+
+                    case CandidateReason.NotInvocable:
+                        // Happens when something like a variable is being invoked, but the variable
+                        // isn't a delegate type.  This may be because the user intends to give 
+                        // this value a delegate type, but hasn't done so yet.  We should still allow
+                        // renaming this reference for now.
+                        return false;
+
+                    case CandidateReason.StaticInstanceMismatch:
+                        // Similar to 'Inaccessible', the code is currently broken, but it's fairly
+                        // clear what the user's intent was.  In this case, we want to update the
+                        // references, even though the code isn't valid.
+                        return true;
+
+                    case CandidateReason.OverloadResolutionFailure:
+                        // Here we were renaming a method, and have a reference location that didn't
+                        // bind properly to any methods.  This case is simply hard to reason about.
+                        // There are times where we might want to rename this, and times when we 
+                        // don't.  As overloading methods is very common, we won't update this location
+                        // as it might update code the user really doesn't want us to be touching.
+                        return false;
+
+                    case CandidateReason.LateBound:
+                        // This is a late bound call, so we should not update this location. 
+                        return false;
+
+                    case CandidateReason.Ambiguous:
+                        // we shoudl not touch amgiguous code.  We have no way to feel confident that
```

Comment:
`We should` (capitalization [you were consistent everywhere else] & spelling), also `ambiguous` spelling